### PR TITLE
If javac issues an error, don't type check any file.

### PR DESCRIPTION
**Pull request by smillst**
This prevents the framework from crashing on ill-formed elements from source files that did not compile.

Fixes https://github.com/typetools/checker-framework/issues/4041.

### CODE REVIEW

**Comment by mernst** [comment]
Please see code review comments in #4354.

File: checker/tests/nullness/java-unsound/Figure1.java [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // Unsound only in Java 8, Java 9+ already gives an error
 // @below-java9-jdk-skip-test
+// @skip-test no need to test for the javac error.
```
Comment (smillst) [inline checker/tests/nullness/java-unsound/Figure1.java:3]: I kept these test files because they are examples in the paper.

**Comment by smillst** [comment]
@mernst this is ready to be reviewed again.  I think we need to pass `-AsuppressWarnings=type.checking.not.run` to the wpiPlumeLib test that is failing, but I can't figure out how to do so.  Can you take a look?

**Review by mernst — CHANGES_REQUESTED** [review]

**Comment by mernst** [comment]
> I think we need to pass `-AsuppressWarnings=type.checking.not.run` to the wpiPlumeLib test that is failing

I agree.

wpi.sh [permits](https://checkerframework.org/manual/#wpi-one) you to pass arguments.  If you run it as `wpi.sh -b SOMEARG ...` then it will run `gradlew compileJava SOMEARG` on the target program.

I don't know how to pass a command-line argument to `gradlew compileJava` so that `-AsuppressWarnings=type.checking.not.run` is passed to `javac`.

@smillst @kelloggm do you have suggestions?  Does this require an enhancement to do-like-javac?

File: framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java [hunk 2]
```diff
@@ -2037,6 +2046,9 @@ public boolean shouldSuppressWarnings(Tree tree, String errKey) {
             throw new BugInCF(
                     "Checker must provide a SuppressWarnings prefix. SourceChecker#getSuppressWarningsPrefixes was not overridden correctly.");
         }
+        if (shouldSuppress(getSuppressWarningsStringsFromOption(), errKey)) {
```
[the remark this answers is not in the document]
Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java:2049]: Yes, I've done so here https://github.com/typetools/checker-framework/pull/4364.

**Review by mernst — APPROVED** [review]
The changes are OK with me, modulo that the tests must pass (which is gradle hacking that @smillst and @kelloggm are better qualified for).

**Comment by kelloggm** [comment]
@mernst @smillst the capability to pass arguments to javac when using dljc's wpi tool existed already, but was undocumented. I've corrected that error here: https://github.com/kelloggm/do-like-javac/pull/17 (the PR is already merged, so you should be able to use it right away). 

You should be able to get this PR working by adding this code to the invocation of `wpi.sh` in `test-wpi-plumelib.sh` anytime after the `--`:
```
--extraJavacArgs='-AsuppressWarnings=type.checking.not.run'
```

Sorry for the inconvenience.

**Pull request merged**